src/functionality/cart.js

```diff
@@ -0,0 +1,90 @@
+const store = require('store2');
+const Orders = require('../collection/resources/orders');
+const Order = require('../collection/objects/order');
+
+class Cart {
+  constructor(request) {
+    this._cart = store.get('cart') || [];
+    this.request = request;
+  }
+
+  _findProductIndex(productId) {
+    return this._cart.findIndex(cartItem => cartItem.productId === productId);
+  }
+
+  _isCartValid() {
+    this._cart.forEach((item) => {
+      if (Object.keys(item).length !== 2 || !item.productId || item.quantity) {
+        return false;
+      }
+    });
+    return true;
+  }
+
+  _saveCart() {
+    store.set('cart', this._cart);
+  }
+
+  set(newCart) {
+    this._cart = newCart;
+  }
+
+  get() {
+    return this._cart;
+  }
+
+  empty() {
+    this._cart = [];
+  }
+
+  addProduct({ productId, quantity }) {
```

Comment:
Maybe add the whole product, then you can retrieve and show product details from the cart?

Reply:
good idea

Reply:
Hm, you do risk showing stale products though. If you just store the ids locally, you could retrieve the whole cart from the API and be sure that the products are up-to-date.